nservicebus/messaging/headers.md

```diff
@@ -73,21 +73,27 @@ A [unique ID for the current message](/nservicebus/messaging/message-identity.md
 
 NServiceBus implements the [Correlation Identifier](http://www.enterpriseintegrationpatterns.com/patterns/messaging/CorrelationIdentifier.html) pattern by using a `Correlation Id` header.
 
-Message correlation connects request messages with their corresponding response messages. The `Correlation Id` of the response message is a `Message Id` of its corresponding request message.
+Message correlation connects request messages with their corresponding response messages. The `Correlation Id` of the response message is the `Correlation Id` of its corresponding request message. Each outgoing message which is sent outside of a message handler will have it's `Correlation Id` set to it's `Message Id`.
```

Comment:
I would say that that behavior is incorrect. In the scenario you described:

> Endpoint A sends Request to Endpoint B
> Endpoint B handles Request and sends Response back to Endpoint A
> Endpoint A handles Response and publishes Event

The Response should have the `Correlation Id` but the Event shouldn't. The Event should have the `Related To` set to the `Message Id` of the Response, and both the Response and the Event should share the same `Conversation Id` that was set on the original Request.